test/mpi/threads/perf/mt_rma_msgrate.c

```diff
@@ -34,7 +34,36 @@ enum RMA_OPs {
     OP_INVALID
 };
 
+enum SYNC_MODE {
+    USE_WIN_FENCE,
+    USE_WIN_LOCK_ALL,
+    SYNC_INVALID,
+};
+
+#define OPEN_WINDOW()                                   \
+    if (sync_mode == USE_WIN_FENCE) {                   \
+        MPI_Win_fence(0, my_win);                       \
+    } else if (sync_mode == USE_WIN_LOCK_ALL) {         \
+        MPI_Win_lock_all(MPI_MODE_NOCHECK, my_win);     \
+    }
+
+#define DO_FENCE()                              \
+    if (sync_mode == USE_WIN_FENCE) {           \
+        MPI_Win_fence(0, my_win);               \
+    }
+
+#define DO_FLUSH()                              \
+    if (sync_mode == USE_WIN_LOCK_ALL) {        \
+        MPI_Win_flush(1, my_win);               \
+    }
+
+#define CLOSE_WINDOW()                          \
+    if (sync_mode == USE_WIN_LOCK_ALL) {        \
+        MPI_Win_unlock_all(my_win);             \
+    }
```

Comment:
The name `WINDOW` is confusing/ambiguous. How about `EPOCH_START` and `EPOCH_END`?

Reply:
Done.